docs/data/date-pickers/fields/fields.md

```diff
@@ -37,25 +37,29 @@ The single input range fields are not supported on the range pickers yet (you ca
 
 The field components have an internal state to update the visible value.
 
-It will only call the `onChange` callback when all the sections of the modified date are filled.
+It will only call the `onChange` callback when:
+
+- all the sections of the modified date are filled
+- all the sections of the modified date are empty (will return `null` for the modified date)
+- all the sections of the modified date were filled, and the user cleaned the content of one section (will return `Invalid date` for modified date)
 
 #### On single element fields
 
-On a single date field (`DateField` / `DateTimeField` / `TimeField`),
-`onChange` will be called if all the sections are filled.
+On a single item field (`DateField` / `DateTimeField` / `TimeField`),
+`onChange` will be called if the date matches one of the condition above.
 
-In the example below, `onChange` will not be fired until the date is fully completed:
+In the example below, `onChange` will be called once when filling the first section and once when filling the last one:
 
 {{"demo": "LifeCycleDateFieldEmpty.js", "defaultCodeOpen": false}}
 
 #### On range fields [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)
 
-On a date range field (`SingleInputDateRangeField` / `MultiInputDateRangeField`),
-`onChange` will be called if all the sections of the date you modified are filled,
-even if some sections of the other date are not filled.
+On a range field (`SingleInputDateRangeField` / `MultiInputDateRangeField` / ... ),
+`onChange` will be called if the date you are modifying is matching one of the condition above,
+even if the other date does not.
 
-In the demo below, changing the value of a start date section will fire `onChange` even if the end date is empty.
-But changing the value of an end date section will not fire `onChange` until the end date is fully completed:
+In the demo below, changing the value of a start date section will call `onChange` even if the end date is empty.
+But when editing the end date, `onChange` will only be called when filling the first section and the last one:
```

Comment:
Are you sure? If I modify a section of the second date, I immediately receive the `Invalid Date`

https://deploy-preview-8082--material-ui-x.netlify.app/x/react-date-pickers/fields/#advanced

Reply:
Indeed, I did not fix that wording, good catch !

Reply:
For me the sentence is correct
When you edit the end date, you have one `onChange` on the 1st edited section (with `Invalid Date`) and one `onChange` on the last edited section (with a valid date)

Reply:
Ok, I understood "first" and "last" according to their position, not "first one to be filled", and "last one to be filled" So it was making no sense 🙈

Maybe the sentence is not needed. It's nice to say that the `onChange` is called even if the end date is empty, but the second part is not related to the range fields. it's the same behavior as simple fields.